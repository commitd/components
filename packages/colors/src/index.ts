export { presetColors } from './preset'
export type { Color } from './preset'
