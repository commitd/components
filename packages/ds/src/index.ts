// component exports
export * from './components'
export * from './utils'

// reexports from ss
export { css, styled } from '@committed/ds-ss'
