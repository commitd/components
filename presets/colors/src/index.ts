export {
  neutralColors,
  preset as presetColors,
  primaryColors,
  semanticColors,
} from './preset'
export type { Color, Prefix, SemanticColor } from './preset'
