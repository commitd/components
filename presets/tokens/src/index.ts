export { preset as presetTokens } from './preset'
