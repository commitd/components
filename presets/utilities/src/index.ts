export { preset as presetUtilities, surfaces } from './preset'
export type { Surface, SurfaceVariants } from './preset'

