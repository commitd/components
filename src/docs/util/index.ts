export * from './Colors'
export * from './Copy'
export * from './SemanticColors'
// export * from './Overview'
export * from './Variants'
