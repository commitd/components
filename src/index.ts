export * from './components'
export * from './theme'
export * from './hooks'

// Reexport
export { styled } from '@material-ui/styles'

// export shortcodes for convenience
import * as shortcodes from './components'
export { shortcodes }
