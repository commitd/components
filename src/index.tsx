export * from './components'
export * from './theme'
export * from './styles'
export * from './util'
// Some explicit exports are required to resolve ambiguity
export { ThemeProvider, colors } from './theme'
export { makeStyles, styled, useTheme, withTheme } from './styles'
export type { ThemeProviderProps } from './theme'
export type { ComponentCreator, WithTheme } from './styles'
