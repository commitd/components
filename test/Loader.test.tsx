import React from 'react'
import { renderLight, renderDark } from './setupTests'
import { Loader } from '../src'
import { Cartesian } from '../stories/util/Cartesian'

it('renders light without crashing', () => {
  const { asFragment } = renderLight(<Loader />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without crashing', () => {
  const { asFragment } = renderDark(<Loader />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders variants without crashing', () => {
  const { asFragment } = renderLight(
    <Cartesian
      component={Loader}
      variant={['draw', 'spin', 'flip', 'scale']}
      loading={[true, false]}
    />
  )
  expect(asFragment()).toMatchSnapshot()
})
