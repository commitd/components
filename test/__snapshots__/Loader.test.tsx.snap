// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without crashing 1`] = `
<DocumentFragment>
  <svg
    class="LogoSvg-root LogoSvg-root makeStyles-color makeStyles-color makeStyles-draw"
    height="1em"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      class="commit_logo"
      d="M215.24 77.76c-.868 3.559-7.539 2.572-10.803 4.426-57.9 16.732-105.793 64.482-122.607 122.209-17.516 57.076-4.33 122.373 34.549 168.197 25.341 30.436 60.348 52.047 98.861 60.888v76.73h81.52v-76.728c61.23-13.633 113.393-61.476 132.248-121.443 18.804-57.048 6.763-122.973-31.692-169.713-25.29-31.406-61.287-54.014-100.595-62.998V2.211h-81.48V77.76zm36.694 78.515c39.604-1.953 78.901 21.962 95.022 58.244 17.615 36.825 9.21 83.858-19.937 112.407-27.57 28.496-72.69 37.623-109.183 22.09-37.152-14.827-62.907-53.607-61.966-93.626-.136-40.346 26.841-78.952 64.788-92.681 9.991-3.841 20.604-5.837 31.276-6.434z"
      fill="#fb0"
      stroke="#fb0"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;

exports[`renders light without crashing 1`] = `
<DocumentFragment>
  <svg
    class="LogoSvg-root LogoSvg-root makeStyles-color makeStyles-color makeStyles-draw"
    height="1em"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      class="commit_logo"
      d="M215.24 77.76c-.868 3.559-7.539 2.572-10.803 4.426-57.9 16.732-105.793 64.482-122.607 122.209-17.516 57.076-4.33 122.373 34.549 168.197 25.341 30.436 60.348 52.047 98.861 60.888v76.73h81.52v-76.728c61.23-13.633 113.393-61.476 132.248-121.443 18.804-57.048 6.763-122.973-31.692-169.713-25.29-31.406-61.287-54.014-100.595-62.998V2.211h-81.48V77.76zm36.694 78.515c39.604-1.953 78.901 21.962 95.022 58.244 17.615 36.825 9.21 83.858-19.937 112.407-27.57 28.496-72.69 37.623-109.183 22.09-37.152-14.827-62.907-53.607-61.966-93.626-.136-40.346 26.841-78.952 64.788-92.681 9.991-3.841 20.604-5.837 31.276-6.434z"
      fill="#fb0"
      stroke="#fb0"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;

exports[`renders variants without crashing 1`] = `
<DocumentFragment>
  <svg
    class="LogoSvg-root LogoSvg-root makeStyles-color makeStyles-color makeStyles-draw"
    height="1em"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      class="commit_logo"
      d="M215.24 77.76c-.868 3.559-7.539 2.572-10.803 4.426-57.9 16.732-105.793 64.482-122.607 122.209-17.516 57.076-4.33 122.373 34.549 168.197 25.341 30.436 60.348 52.047 98.861 60.888v76.73h81.52v-76.728c61.23-13.633 113.393-61.476 132.248-121.443 18.804-57.048 6.763-122.973-31.692-169.713-25.29-31.406-61.287-54.014-100.595-62.998V2.211h-81.48V77.76zm36.694 78.515c39.604-1.953 78.901 21.962 95.022 58.244 17.615 36.825 9.21 83.858-19.937 112.407-27.57 28.496-72.69 37.623-109.183 22.09-37.152-14.827-62.907-53.607-61.966-93.626-.136-40.346 26.841-78.952 64.788-92.681 9.991-3.841 20.604-5.837 31.276-6.434z"
      fill="#fb0"
      stroke="#fb0"
      stroke-width="4"
    />
  </svg>
  <svg
    class="LogoSvg-root LogoSvg-root makeStyles-color makeStyles-color"
    height="1em"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      class="commit_logo"
      d="M215.24 77.76c-.868 3.559-7.539 2.572-10.803 4.426-57.9 16.732-105.793 64.482-122.607 122.209-17.516 57.076-4.33 122.373 34.549 168.197 25.341 30.436 60.348 52.047 98.861 60.888v76.73h81.52v-76.728c61.23-13.633 113.393-61.476 132.248-121.443 18.804-57.048 6.763-122.973-31.692-169.713-25.29-31.406-61.287-54.014-100.595-62.998V2.211h-81.48V77.76zm36.694 78.515c39.604-1.953 78.901 21.962 95.022 58.244 17.615 36.825 9.21 83.858-19.937 112.407-27.57 28.496-72.69 37.623-109.183 22.09-37.152-14.827-62.907-53.607-61.966-93.626-.136-40.346 26.841-78.952 64.788-92.681 9.991-3.841 20.604-5.837 31.276-6.434z"
      fill="#fb0"
      stroke="#fb0"
      stroke-width="4"
    />
  </svg>
  <svg
    class="LogoSvg-root LogoSvg-root makeStyles-color makeStyles-color makeStyles-spin"
    height="1em"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      class="commit_logo"
      d="M215.24 77.76c-.868 3.559-7.539 2.572-10.803 4.426-57.9 16.732-105.793 64.482-122.607 122.209-17.516 57.076-4.33 122.373 34.549 168.197 25.341 30.436 60.348 52.047 98.861 60.888v76.73h81.52v-76.728c61.23-13.633 113.393-61.476 132.248-121.443 18.804-57.048 6.763-122.973-31.692-169.713-25.29-31.406-61.287-54.014-100.595-62.998V2.211h-81.48V77.76zm36.694 78.515c39.604-1.953 78.901 21.962 95.022 58.244 17.615 36.825 9.21 83.858-19.937 112.407-27.57 28.496-72.69 37.623-109.183 22.09-37.152-14.827-62.907-53.607-61.966-93.626-.136-40.346 26.841-78.952 64.788-92.681 9.991-3.841 20.604-5.837 31.276-6.434z"
      fill="#fb0"
      stroke="#fb0"
      stroke-width="4"
    />
  </svg>
  <svg
    class="LogoSvg-root LogoSvg-root makeStyles-color makeStyles-color"
    height="1em"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      class="commit_logo"
      d="M215.24 77.76c-.868 3.559-7.539 2.572-10.803 4.426-57.9 16.732-105.793 64.482-122.607 122.209-17.516 57.076-4.33 122.373 34.549 168.197 25.341 30.436 60.348 52.047 98.861 60.888v76.73h81.52v-76.728c61.23-13.633 113.393-61.476 132.248-121.443 18.804-57.048 6.763-122.973-31.692-169.713-25.29-31.406-61.287-54.014-100.595-62.998V2.211h-81.48V77.76zm36.694 78.515c39.604-1.953 78.901 21.962 95.022 58.244 17.615 36.825 9.21 83.858-19.937 112.407-27.57 28.496-72.69 37.623-109.183 22.09-37.152-14.827-62.907-53.607-61.966-93.626-.136-40.346 26.841-78.952 64.788-92.681 9.991-3.841 20.604-5.837 31.276-6.434z"
      fill="#fb0"
      stroke="#fb0"
      stroke-width="4"
    />
  </svg>
  <svg
    class="LogoSvg-root LogoSvg-root makeStyles-color makeStyles-color makeStyles-flip"
    height="1em"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      class="commit_logo"
      d="M215.24 77.76c-.868 3.559-7.539 2.572-10.803 4.426-57.9 16.732-105.793 64.482-122.607 122.209-17.516 57.076-4.33 122.373 34.549 168.197 25.341 30.436 60.348 52.047 98.861 60.888v76.73h81.52v-76.728c61.23-13.633 113.393-61.476 132.248-121.443 18.804-57.048 6.763-122.973-31.692-169.713-25.29-31.406-61.287-54.014-100.595-62.998V2.211h-81.48V77.76zm36.694 78.515c39.604-1.953 78.901 21.962 95.022 58.244 17.615 36.825 9.21 83.858-19.937 112.407-27.57 28.496-72.69 37.623-109.183 22.09-37.152-14.827-62.907-53.607-61.966-93.626-.136-40.346 26.841-78.952 64.788-92.681 9.991-3.841 20.604-5.837 31.276-6.434z"
      fill="#fb0"
      stroke="#fb0"
      stroke-width="4"
    />
  </svg>
  <svg
    class="LogoSvg-root LogoSvg-root makeStyles-color makeStyles-color"
    height="1em"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      class="commit_logo"
      d="M215.24 77.76c-.868 3.559-7.539 2.572-10.803 4.426-57.9 16.732-105.793 64.482-122.607 122.209-17.516 57.076-4.33 122.373 34.549 168.197 25.341 30.436 60.348 52.047 98.861 60.888v76.73h81.52v-76.728c61.23-13.633 113.393-61.476 132.248-121.443 18.804-57.048 6.763-122.973-31.692-169.713-25.29-31.406-61.287-54.014-100.595-62.998V2.211h-81.48V77.76zm36.694 78.515c39.604-1.953 78.901 21.962 95.022 58.244 17.615 36.825 9.21 83.858-19.937 112.407-27.57 28.496-72.69 37.623-109.183 22.09-37.152-14.827-62.907-53.607-61.966-93.626-.136-40.346 26.841-78.952 64.788-92.681 9.991-3.841 20.604-5.837 31.276-6.434z"
      fill="#fb0"
      stroke="#fb0"
      stroke-width="4"
    />
  </svg>
  <svg
    class="LogoSvg-root LogoSvg-root makeStyles-color makeStyles-color makeStyles-scale"
    height="1em"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      class="commit_logo"
      d="M215.24 77.76c-.868 3.559-7.539 2.572-10.803 4.426-57.9 16.732-105.793 64.482-122.607 122.209-17.516 57.076-4.33 122.373 34.549 168.197 25.341 30.436 60.348 52.047 98.861 60.888v76.73h81.52v-76.728c61.23-13.633 113.393-61.476 132.248-121.443 18.804-57.048 6.763-122.973-31.692-169.713-25.29-31.406-61.287-54.014-100.595-62.998V2.211h-81.48V77.76zm36.694 78.515c39.604-1.953 78.901 21.962 95.022 58.244 17.615 36.825 9.21 83.858-19.937 112.407-27.57 28.496-72.69 37.623-109.183 22.09-37.152-14.827-62.907-53.607-61.966-93.626-.136-40.346 26.841-78.952 64.788-92.681 9.991-3.841 20.604-5.837 31.276-6.434z"
      fill="#fb0"
      stroke="#fb0"
      stroke-width="4"
    />
  </svg>
  <svg
    class="LogoSvg-root LogoSvg-root makeStyles-color makeStyles-color"
    height="1em"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      class="commit_logo"
      d="M215.24 77.76c-.868 3.559-7.539 2.572-10.803 4.426-57.9 16.732-105.793 64.482-122.607 122.209-17.516 57.076-4.33 122.373 34.549 168.197 25.341 30.436 60.348 52.047 98.861 60.888v76.73h81.52v-76.728c61.23-13.633 113.393-61.476 132.248-121.443 18.804-57.048 6.763-122.973-31.692-169.713-25.29-31.406-61.287-54.014-100.595-62.998V2.211h-81.48V77.76zm36.694 78.515c39.604-1.953 78.901 21.962 95.022 58.244 17.615 36.825 9.21 83.858-19.937 112.407-27.57 28.496-72.69 37.623-109.183 22.09-37.152-14.827-62.907-53.607-61.966-93.626-.136-40.346 26.841-78.952 64.788-92.681 9.991-3.841 20.604-5.837 31.276-6.434z"
      fill="#fb0"
      stroke="#fb0"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;
